api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceService.java

```diff
@@ -0,0 +1,57 @@
+package org.pmiops.workbench.db.dao;
+
+import java.util.logging.Logger;
+import org.pmiops.workbench.db.model.Workspace;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * Workspace manipulation and shared business logic which can't be represented by automatic query
+ * generation in WorkspaceDao or @Query annotations.
+ *
+ * TODO(RW-215) Add versioning to detect/prevent concurrent edits.
+ */
+@Service
+public class WorkspaceService {
+  private static final Logger log = Logger.getLogger(WorkspaceService.class.getName());
+
+  /**
+   * Clients wishing to use the auto-generated methods from the DAO interface may directly access
+   * it here.
+   */
+  public final WorkspaceDao dao;
+
+  @Autowired
+  public WorkspaceService(WorkspaceDao workspaceDao) {
+    this.dao = workspaceDao;
+  }
+
+  public Workspace getRequired(String ns, String id) {
+    Workspace workspace = dao.get(ns, id);
+    if (workspace == null) {
+      throw new NotFoundException(String.format("Workspace %s/%s not found.", ns, id));
+    }
+    return workspace;
+  }
+
+  // FIXME @Version instead? Bean instantiation v. @Transactional?
```

Comment:
I assume we'll tackle the @Version stuff in a follow-up?

Reply:
That's my current thought, curious to hear others' opinions. I think it's a relatively clean separation, and I'd like to let Brew make other API changes and/or use the API. But it is a significant broken-ness in the current state.